pyomo/neos/tests/test_neos.py

```diff
@@ -90,6 +88,177 @@ def test_kestrel_plugin(self):
+
+    def test_loqo(self):
+        self._run('loqo')
+
+    def test_minlp(self):
+        self._run('minlp')
+
+    def test_minos(self):
+        self._run('minos')
+
+    # This solver doesn't generate a solution file
+    # when the presolver finds the solution
+    #def test_minto(self):
+    #    self._run('minto')
+
+    def test_mosek(self):
+        self._run('mosek')
+
+    # This solver doesn't give the same answer for
+    # min f(x) and max -f(x)
+    #def test_ooqp(self):
+    #    self._run('ooqp')
+
+    # The simple tests aren't complementarity 
+    # problems
+    #def test_path(self):
+    #    self._run('path')
+
+    def test_snopt(self):
+        self._run('snopt')
+
+    # This test only handles minimization problems
+    #def test_raposa(self):
+    #    self._run('raposa')
+
+    def test_lgo(self):
+        self._run('lgo')
+
+
+@unittest.category('neos')
+@unittest.skipIf(not neos_available, "Cannot make connection to NEOS server")
+class TestSolvers_script_max(TestSolvers_script_min):
+
+    def _model(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var(bounds=(0,1), initialize=0)
+        m.c = pyo.Constraint(expr=m.x <= 0.5)
+        m.obj = pyo.Objective(expr=2*m.x, sense=pyo.maximize)
+        return m
+
+
+@unittest.category('neos')
+@unittest.skipIf(not neos_available, "Cannot make connection to NEOS server")
+class TestSolvers_cmd_min(TestSolvers_script_min):
+
+    filename = 't2.py'
+    objective = -1
+
+    def _run(self, opt):
+        results = os.path.join(currdir, 'result.json')
+        args = [
+            os.path.join(currdir,self.filename),
+            '--solver-manager=neos',
+            '--solver=%s' % opt,
+            '--logging=quiet',
+            '--save-results=%s' % results,
+            '-c'
+            ]
+        try:
+            output = main.run(args)
+            self.assertEqual(output.errorcode, 0)
+
+            with open(results) as FILE:
+                data = json.load(FILE)
+            #print(json.dumps(data, indent=4))
+
+            self.assertEqual(
+                data['Solver'][0]['Status'], 'ok')
+            self.assertAlmostEqual(
+                data['Solution'][1]['Status'], 'optimal')
+            if 'x' in data['Solution'][1]['Variable']:
+                self.assertAlmostEqual(
+                    data['Solution'][1]['Variable']['x']['Value'], 0.5)
+            else:
+                self.fail("Expected nonzero solution variables")
+            if 'o' in data['Solution'][1]['Objective']:
+                self.assertAlmostEqual(
+                    data['Solution'][1]['Objective']['o']['Value'], self.objective)
+        finally:
+            cleanup()
+            if os.path.exists(results):
+                os.remove(results)
+
+
+@unittest.category('neos')
+@unittest.skipIf(not neos_available, "Cannot make connection to NEOS server")
+class TestSolvers_cmd_max(TestSolvers_cmd_min):
+
+    filename = 't1.py'
+    objective = 1
```

Comment:
Is this test complete? It doesn't have anything besides a filename and objective.

Reply:
Correct.